### PR TITLE
chore: replace `oneapi-src` with `uxlfoundation`

### PR DESCRIPTION
**Pull request by homksei**
## Description

This Pull Request updates the organization name (from `oneapi-src` to `uxlfoundation`) in all relevant entries within the repository following the transfer of the repository to a new organization.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
Should be addressed by this other PR:
https://github.com/uxlfoundation/oneDAL/pull/3004
which also updates the links to sklearnex.

**Comment by homksei** [comment]
> Should be addressed by this other PR: #3004 which also updates the links to sklearnex.

As far as I can see in your PR, you are updating the Doxygen documentation, and it is not related to replacing occurrences of oneapi-src with uxlfoundation.

In any case, you can always cherry-pick the changes from this PR (or make them yourself) into your own. Then this can be closed.

**Comment by david-cortes-intel** [comment]
> > Should be addressed by this other PR: #3004 which also updates the links to sklearnex.
> 
> As far as I can see in your PR, you are updating the Doxygen documentation, and it is not related to replacing occurrences of oneapi-src with uxlfoundation.
> 
> In any case, you can always cherry-pick the changes from this PR (or make them yourself) into your own. Then this can be closed.

Sorry, wrong link, should have been this one: https://github.com/uxlfoundation/oneDAL/pull/3003

**Comment by ethanglaser** [comment]
https://github.com/uxlfoundation/oneDAL/pull/3003 is closer to completion
